### PR TITLE
fix(x11): use normal malloc for frame buffer allocation

### LINKED ISSUE #6347: X11 driver not correctly working (or I am doing it wrong)
### LVGL version

v9.1.0

### What happened?

The X11 driver used for standalone desktop mode does not work correctly (or I am doing it wrong). Depending on the initial resolution, it does either work, segfault, or the function does not even return:

lvgl is initialized as follows:

```c
lv_init();
lv_display_t *disp = lv_x11_window_create("myname", width, height);
lv_x11_inputs_create(disp, NULL);
```

Depending on the resolution in `width` and `height` different things happen:

* 100x64: works as expected
* 256x347: segfaults later in program when calling lv_obj_align()
* 800x600: lv_x11_window_create() does not return

Here ist the code to draw the screen:

```c
lv_obj_set_style_bg_color(lv_screen_active(), lv_color_hex(0x003a57), LV_PART_MAIN);
lv_obj_t *label = lv_label_create(lv_screen_active());
lv_label_set_text(label, "Hello world");
lv_obj_set_style_text_color(lv_screen_active(), lv_color_hex(0xffffff), LV_PART_MAIN);
lv_obj_align(label, LV_ALIGN_CENTER, 0, 0);
```

### How to reproduce?

```c
lv_init();
lv_display_t *disp = lv_x11_window_create("myname", width, height);
lv_x11_inputs_create(disp, NULL);
lv_obj_set_style_bg_color(lv_screen_active(), lv_color_hex(0x003a57), LV_PART_MAIN);
lv_obj_t *label = lv_label_create(lv_screen_active());
lv_label_set_text(label, "Hello world");
lv_obj_set_style_text_color(lv_screen_active(), lv_color_hex(0xffffff), LV_PART_MAIN);
lv_obj_align(label, LV_ALIGN_CENTER, 0, 0);
```

### PR DESCRIPTION
**Pull request by kisvegabor**
fixes #6347

### Description of the feature or fix

A clear and concise description of what the bug or new feature is.

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.12](https://github.com/szepeviktor/astyle/releases/tag/v3.4.12) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Review by liamHowatt — COMMENTED** [review]
```c
    lv_free(xd->buffer[0]);
    if(LV_X11_DOUBLE_BUFFER) {
        lv_free(xd->buffer[1]);
    }
```
Please update this part to use `free`.

File: src/drivers/x11/lv_x11_display.c [hunk 1]
```diff
@@ -378,8 +378,8 @@ lv_display_t * lv_x11_window_create(char const * title, int32_t hor_res, int32_t
     if(LV_X11_RENDER_MODE == LV_DISPLAY_RENDER_MODE_PARTIAL) {
         sz_buffers /= 10;
     }
-    xd->buffer[0] = lv_malloc(sz_buffers);
-    xd->buffer[1] = (LV_X11_DOUBLE_BUFFER ? lv_malloc(sz_buffers) : NULL);
+    xd->buffer[0] = malloc(sz_buffers);
```
Comment (FASTSHIFT) [inline src/drivers/x11/lv_x11_display.c:381]: Should use `free` to release the corresponding `xd->buffer` instead of `lv_free`.

**Comment by kisvegabor** [comment]
Nice catch, thanks! Fixed.

**Review by C47D — APPROVED** [review]

**Review by FASTSHIFT — DISMISSED** [review]

**Review by liamHowatt — COMMENTED** [review]
There's a `lv_realloc` here too. Please update that to `realloc`.

**Comment by kisvegabor** [comment]
Ah, I feel ashamed... Sorry...

**Review by FASTSHIFT — APPROVED** [review]

**Review by liamHowatt — APPROVED** [review]
It's an easy thing to miss!

**Pull request merged**